File: samples/client/3_1_0_unit_test/java/src/main/java/org/openapijsonschematools/client/schemas/validation/ValidationData.java
package org.openapijsonschematools.client.schemas.validation;

import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.List;

public record ValidationData(
    JsonSchema schema,
    @Nullable Object arg,
    ValidationMetadata validationMetadata,
    @Nullable List<PathToSchemasMap> containsPathToSchemas,
    @Nullable PathToSchemasMap patternPropertiesPathToSchemas,
    @Nullable PathToSchemasMap ifPathToSchemas) {
}

File: samples/client/petstore/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/ValidationData.kt
package org.openapijsonschematools.client.schemas.validation

data class ValidationData(
    val schema: JsonSchema<*>,
    val arg: Any?,
    val validationMetadata: ValidationMetadata,
    val containsPathToSchemas: List<PathToSchemasMap>?,
    val patternPropertiesPathToSchemas: PathToSchemasMap?,
    val ifPathToSchemas: PathToSchemasMap?,
    val knownPathToSchemas: PathToSchemasMap?
)